controllers/barbican_controller.go

```diff
@@ -919,11 +949,53 @@ func (r *BarbicanReconciler) reconcileInit(
 	}
 	instance.Status.Conditions.MarkTrue(condition.DBSyncReadyCondition, condition.DBSyncReadyMessage)
 
+	//
+	// run Barbican p11-prep if needed
+	//
+	if slices.Contains(instance.Spec.EnabledSecretStores, "pkcs11") {
```

Comment:
We should have an else here, update the status condition to true with a message like:

"P11 Prep job not run"

This is why it's failing to create a deployment after kuttl tests run, since by not setting it, "Ready" condition is also not set in [1], and consequently the openstack control plane is also not ready.

[1] https://github.com/openstack-k8s-operators/barbican-operator/pull/168/files#diff-cb25cf601c065f6057f4ae086a63e5796d9790074eb5b842f2b9169fed00cd58L447